sdf.md

```diff
@@ -139,7 +139,46 @@ The target namespace is the namespace into which the defined terms are added. Th
 
 For example if the default namespace in the example above is "foo", then you could use "temperatureData" to refer to the property defined at the URI:
 ```
-https://example.com/#temperatureData
+https://example.com/#odmData/temperatureData
+```
+
+## Optionality using the keyword "required"
+
+The keyword "required" is provided to apply a constraint for which definitions are mandatory in an instance conforming to a particular definition in which the constraint appears.
+
+The value of "required" is an array of reference objects, each of which contains a JSON pointer to one required definition.
```

Comment:
I actually found the use of member names instead of JSON pointers more convenient. Was there a strong reason to use JSON pointers instead?

Reply:
We didn't want to require every name to be unique, so the path or some other differentiator is needed. 

Originally we proposed a dot notation, e.g. odm:BinarySwitch.State but there would need to be some mechanism to unpack the definition to know which term represents the odmObject, odmProperty, etc.

JSON pointer gives us a well known syntax for this, and we are already using JSON pointer for common definitions in the SDF file, ala JSON Schema.

If we compare some different approaches, we have:

Unique names:
odm:BinarySwitchObject
odm:BinarySwitchStateProperty

Dot notation:
odm:BInarySwitch.State

JSON Pointer:
odm:odmObject/BinarySwitch/odmProperty/State

Is there some other way we can construct references that would work out better?

Reply:
Not sure of better, but a different option would be to have separate “required” block for each of the Action, Events, and Properties blocks. This would also match how JSON schema uses “required”.

Reply:
F2F-4: Agreed to use json pointer syntax (RFC6901 and draft for relative pointers), but that means there will need to be special processing during reference resolution of the pointer (it can be locally resolved or resolved to an external file). Add "odmRequired" for syntax representing the ODM elements that are required (JSON schema "required" still is used in JSON schemas).

Reply:
F2F4: Change "required" to "odmRequired".